src/qt/paymentserver.cpp

```diff
@@ -414,7 +422,71 @@ void PaymentServer::handleURIOrFile(const QString& s)
 #else
         QUrlQuery uri((QUrl(s)));
 #endif
-        if (uri.hasQueryItem("r")) // payment request URI
+        if (uri.hasQueryItem("m") && uri.hasQueryItem("a")) // payment request URI
+        {
+            QByteArray temp;
+            temp.append(uri.queryItemValue("m"));
+            temp.append(uri.queryItemValue("a"));
+            QString decoded = QUrl::fromPercentEncoding(temp);
+            QUrl fetchUrl(decoded, QUrl::StrictMode);
+
+            if (!model)
+                return;
+
+            CNavCoinAddress addr(uri.queryItemValue("a").toStdString());
+            if (!addr.IsValid())
+            {
+                Q_EMIT message(tr("Verify address"), tr("The entered address is invalid.") + QString(" ") + tr("Please check the address and try again."),
+                               CClientUIInterface::ICON_WARNING);
+                return;
+            }
+            CKeyID keyID;
+            if (!addr.GetKeyID(keyID))
+            {
+                Q_EMIT message(tr("Verify address"), tr("The entered address does not refer to a key.") + QString(" ") + tr("Please check the address and try again."),
+                               CClientUIInterface::ICON_WARNING);
+                return;
+            }
+
+            WalletModel::UnlockContext ctx(model->requestUnlock());
```

Comment:
General comment, I could be blind but it doesn't look like you're checking if:
- the string to sign is too long
- the string to sign is too short (i.e. length = 0)
- the url is valid